shared/src/main/java/com/couchbase/lite/PropertiesImpl.java

```diff
@@ -177,43 +124,22 @@ public Blob getBlob(String key) {
 
     @Override
     public Date getDate(String key) {
-        if (properties != null) {
-            Object obj = properties.get(key);
-            if (obj != null && obj instanceof String)
-                return DateUtils.fromJson((String) obj);
-            else
-                return null;
-        } else {
-            if (root == null)
-                return null;
-            else {
-                FLValue flvalue = root.get(key);
-                if (flvalue.getType() == kFLString)
-                    return DateUtils.fromJson(flvalue.asString());
-                else
-                    return null;
-            }
+        if (properties != null)
+            return DateUtils.fromJson(cast(properties.get(key), String.class));
+        else {
+            FLValue flvalue = getValueFromRoot(key);
+            return flvalue == null || flvalue.getType() != kFLString
+                    ? null : DateUtils.fromJson(flvalue.asString());
```

Comment:
This can be done by `return DateUtils.fromJson(getString(key))` so that the get string code doesn't get repeated here.